### PR TITLE
Implemented  ProvisionerGrantBucketAccess call  for  driver

### PR DESCRIPTION
**Pull request by sajanjswl**
Signed-off-by: Sajan Jaiswal <sjnjaiswal@gmail.com>

### CODE REVIEW

File: examples/sample-provisioner/driver/driverserver.go [hunk 1]
```diff
@@ -76,7 +83,41 @@ func (ds *DriverServer) ProvisionerDeleteBucket(ctx context.Context, req *cosi.P
...
+	}
+
+	if err := ds.S3AdminClient.SetPolicy(context.Background(), "write-get", req.Principal, false); err != nil {
+		klog.Error("failed to set policy", err)
+		return nil, err
+	}
+
+	return &cosi.ProvisionerGrantBucketAccessResponse{
+		Principal:               req.Principal,
+		CredentialsFileContents: fmt.Sprintf("%s \n %s", creds.AccessKey, creds.SecretKey),
```
Comment (wlan0) [inline examples/sample-provisioner/driver/driverserver.go:118]: [default]
aws_access_key=
aws_acces_secret_key=

File: examples/sample-provisioner/driver/driverserver.go [hunk 2]
```diff
@@ -76,7 +83,41 @@ func (ds *DriverServer) ProvisionerDeleteBucket(ctx context.Context, req *cosi.P
...
+		return nil, err
+	}
+
+	// Create policy
+	p := iampolicy.Policy{
+		Version: iampolicy.DefaultVersion,
+		Statements: []iampolicy.Statement{
+			iampolicy.NewStatement(
+				policy.Allow,
+				iampolicy.NewActionSet(iampolicy.GetObjectAction, iampolicy.PutObjectAction),
```
Comment (wlan0) [inline examples/sample-provisioner/driver/driverserver.go:100]: * policy for all S3 operations

File: examples/sample-provisioner/driver/driverserver.go [hunk 3]
```diff
@@ -76,7 +83,41 @@ func (ds *DriverServer) ProvisionerDeleteBucket(ctx context.Context, req *cosi.P
...
+
+	if err := ds.S3AdminClient.SetPolicy(context.Background(), "write-get", req.Principal, false); err != nil {
+		klog.Error("failed to set policy", err)
+		return nil, err
+	}
+
+	return &cosi.ProvisionerGrantBucketAccessResponse{
+		Principal:               req.Principal,
+		CredentialsFileContents: fmt.Sprintf("%s \n %s", creds.AccessKey, creds.SecretKey),
+		CredentialsFilePath:     ".minio/credentials",
```
Comment (wlan0) [inline examples/sample-provisioner/driver/driverserver.go:119]: .aws/credentials

**Comment by sajanjswl** [comment]
fixed

**Pull request merged**